internal/database/database.go

```diff
@@ -19,7 +20,33 @@ func (c *Config) SetConnector(connector dialect.Connector) {
 	c.connector = connector
 }
 
-func Connect(config Config, useAdmin bool) (*sql.DB, error) {
+type DB struct {
+	*sql.DB
+	database dialect.Database
+}
+
+// SetDatabase is used for testing purposes
+func (db *DB) SetDatabase(database dialect.Database) {
+	db.database = database
+}
+
+func (db *DB) DatabaseName() string {
+	return db.database.DatabaseName()
+}
+
+func (db *DB) Username() string {
+	return db.database.Username()
+}
+
+func (db *DB) Type() string {
+	return db.database.Type()
+}
+
+func (db *DB) Timetravel(d time.Duration) string {
+	return db.database.Timetravel(d)
+}
```

Comment:
Why don't we just embed the dialect.Database interface?

Reply:
you are right